### PR TITLE
add gha and linting

### PR DESCRIPTION
**Pull request by acharb**
adding gha to automate tests, and linting

[linting ticket](https://stellarorg.atlassian.net/browse/WAL-1046?atlOrigin=eyJpIjoiY2VlZjc4ZDA5MGNhNDQ3ODk1ODNmZWRjYTYxZmE4NmUiLCJwIjoiaiJ9)

[automated test ticket](https://stellarorg.atlassian.net/browse/WAL-1068?atlOrigin=eyJpIjoiMWM4MTIwZWUzMjNiNDIwY2FlYmVjMDNhNDg0MGZjOWUiLCJwIjoiaiJ9)

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    project: ["tsconfig.json", "examples/tsconfig.json", "test/tsconfig.json"],
+    sourceType: "module",
+  },
+  plugins: ["@typescript-eslint", "jsdoc"],
+  extends: [
+    "plugin:@typescript-eslint/recommended-requiring-type-checking",
+    "plugin:@typescript-eslint/recommended",
+    "prettier",
+  ],
+  rules: {
```
[the remark this answers is not in the document]
Comment (Ifropc) [reply .eslintrc.js:14]: Do we have a standard eslint config for Stellar projects?

File: package.json [hunk 2]
```diff
@@ -38,6 +41,7 @@
   },
   "dependencies": {
     "axios": "^1.4.0",
+    "eslint-plugin-jsdoc": "^46.8.2",
```
Comment (Ifropc) [inline package.json:44]: Why is it not a dev dependency?

File: package.json [hunk 3]
```diff
@@ -49,11 +53,18 @@
     "utility-types": "^3.10.0"
   },
   "scripts": {
+    "lint": "eslint . --ext .ts",
     "prepare": "husky install",
     "test": "jest --watchAll",
+    "test:ci": "jest --ci",
```
Comment (Ifropc) [inline package.json:59]: Not very familiar with jest, what's the difference between `watchAll` and `ci`?

**Review by Ifropc — COMMENTED** [review]
LGTM just couple questions

File: .eslintrc.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (acharb) [reply .eslintrc.js:14]: the only one I'm aware of that product conventions one linked there which we're using in all of our frontend projects (freighter, vibrant, lab etc). I copied most of those configs w/o the stuff that didn't apply (like react stuff)

cc @piyalbasu if there's any eslint config I'm not aware of to use here

File: package.json (hunk 3 above)
Comment (acharb) [reply package.json:59]: it lets jest know it's running in a ci environment (according to jest [docs](https://jestjs.io/docs/cli#--ci), it looks like it only affects snapshots, which we're not doing here). We're using the same thing in our other projects, all the tests still run

File: package.json (hunk 2 above)
Comment (acharb) [reply package.json]: whoops good catch
https://github.com/stellar/typescript-wallet-sdk/pull/72/commits/038808837b722268e69d421cfbab4fba91eca682

**Pull request merged**